java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java

```diff
@@ -21,480 +21,322 @@
 import org.apache.arrow.vector.holders.ComplexHolder;
-import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
+import org.apache.arrow.vector.types.Types.MinorType;
 import org.apache.arrow.vector.types.pojo.ArrowType;
-import org.apache.arrow.vector.types.pojo.ArrowType.Struct;
-import org.apache.arrow.vector.types.pojo.DictionaryEncoding;
-import org.apache.arrow.vector.types.pojo.FieldType;
 import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
 import org.apache.arrow.vector.util.CallBack;
-import org.apache.arrow.vector.util.OversizedAllocationException;
+import org.apache.arrow.vector.util.JsonStringHashMap;
 import org.apache.arrow.vector.util.TransferPair;
 
-public class MapVector extends NonNullableMapVector implements FieldVector {
+public class MapVector extends AbstractMapVector {
+  //private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(MapVector.class);
 
   public static MapVector empty(String name, BufferAllocator allocator) {
-    FieldType fieldType = FieldType.nullable(Struct.INSTANCE);
+    FieldType fieldType = new FieldType(false, ArrowType.Struct.INSTANCE, null, null);
     return new MapVector(name, allocator, fieldType, null);
   }
 
-  private final NullableMapReaderImpl reader = new NullableMapReaderImpl(this);
-  private final NullableMapWriter writer = new NullableMapWriter(this);
-
-  protected ArrowBuf validityBuffer;
-  private int validityAllocationSizeInBytes;
+  private final SingleMapReaderImpl reader = new SingleMapReaderImpl(this);
+  protected final FieldType fieldType;
+  public int valueCount;
 
   // deprecated, use FieldType or static constructor instead
   @Deprecated
   public MapVector(String name, BufferAllocator allocator, CallBack callBack) {
-    this(name, allocator, FieldType.nullable(ArrowType.Struct.INSTANCE), callBack);
+    this(name, allocator, new FieldType(false, ArrowType.Struct.INSTANCE, null, null), callBack);
   }
 
-  // deprecated, use FieldType or static constructor instead
-  @Deprecated
-  public MapVector(String name, BufferAllocator allocator, DictionaryEncoding dictionary, CallBack callBack) {
-    this(name, allocator, new FieldType(true, ArrowType.Struct.INSTANCE, dictionary, null), callBack);
+  public MapVector(String name, BufferAllocator allocator, FieldType fieldType, CallBack callBack) {
+    super(name, allocator, callBack);
+    this.fieldType = checkNotNull(fieldType);
+    this.valueCount = 0;
   }
 
-  public MapVector(String name, BufferAllocator allocator, FieldType fieldType, CallBack callBack) {
-    super(name, checkNotNull(allocator), fieldType, callBack);
-    this.validityBuffer = allocator.getEmpty();
-    this.validityAllocationSizeInBytes = BitVectorHelper.getValidityBufferSize(BaseValueVector.INITIAL_VALUE_ALLOCATION);
+  @Override
+  public FieldReader getReader() {
+    return reader;
+  }
+
+  transient private MapTransferPair ephPair;
```

Comment:
`ephPair` seems wrong

Reply:
how so, you mean the name?

Reply:
keep in mind the diff is showing the result of renaming `NonNullableMapVector` to `MapVector` and none of the code was actually changed, just the class name

Reply:
Oh sorry I should be clear - the variable name `ephPair` seems wrong - I don't know if it has always been `ephPair` or somehow changed during refactoring. Either way, seems worth fixing.

Reply:
Oh ok, I'm not sure of the reason for the name, this is how it was previously to ARROW-1710.  I'd prefer to keep the changes here to that scope so we can be sure they are restored correctly.

Reply:
Ok, sounds good.